src/Microsoft.OpenApi/Writers/OpenApiWriterExtensions.cs

```diff
@@ -132,6 +224,33 @@ public static void WriteProperty<T>(this IOpenApiWriter writer, string name, T v
             }
         }
 
+        /// <summary>
+        /// Write the optional Open API object/element.
+        /// </summary>
+        /// <typeparam name="T">The Open API element type. <see cref="IOpenApiElement"/></typeparam>
+        /// <param name="writer">The Open API writer.</param>
+        /// <param name="name">The property name.</param>
+        /// <param name="value">The property value.</param>
+        /// <param name="action">The proprety value writer action.</param>
+        public static async Task WriteOptionalObjectAsync<T>(
+            this IOpenApiWriter writer,
+            string name,
+            T value,
+            Action<IOpenApiWriter, T> action)
```

Comment:
Should be `Func<IOpenApiWriter, T, Task>` so that the caller can provide something that calls asynchronous methods of `IOpenApiWriter`.